packages/test/test-end-to-end-tests/src/test/dataStoresNested.spec.ts

```diff
@@ -0,0 +1,194 @@
+		registryEntries: [[dataStoreFactory.type, Promise.resolve(dataStoreFactory)]],
+		runtimeOptions,
+	});
+
+	async function addContainer(container: IContainer) {
+		containers.push(container);
+		const dataStores = (await container.getEntryPoint()) as IDataStores;
+
+		await provider.ensureSynchronized();
+
+		return { container, dataStores };
+	}
+
+	const createContainer = async () => {
+		const container = await provider.createContainer(runtimeFactory);
+		return addContainer(container);
+	};
+
+	async function addContainerInstance() {
+		const container = await provider.loadContainer(runtimeFactory);
+		return addContainer(container);
+	}
+
+	beforeEach("getTestObjectProvider", async () => {
+		const driver = new LocalServerTestDriver();
+		const registry = [];
+		seed = 1; // Every test is independent from another test!
+
+		provider = new TestObjectProvider(
+			Loader,
+			driver,
+			() =>
+				new TestContainerRuntimeFactory(
+					"@fluid-experimental/test-dataStores",
+					new TestFluidObjectFactory(registry),
+				),
+		);
+		// syncSummarizer: true
+		provider.resetLoaderContainerTracker(true); // syncSummarizerClients
+
+		loader = provider.createLoader([[provider.defaultCodeDetails, runtimeFactory]]);
+	});
+
+	afterEach(() => {
+		provider.reset();
+		for (const container of containers) {
+			container.close();
+		}
+		containers = [];
+		summaryCollection = undefined;
+		summarizer = undefined;
+		loader = undefined;
+	});
+
+	async function waitForSummary(): Promise<string> {
+		// ensure that all changes made it through
+		await provider.ensureSynchronized();
+
+		assert(summaryCollection !== undefined, "summary setup properly");
+		// create promise before we call summarizeNow, as otherwise we might miss summary and will wait
+		// forever for next one to happen
+		const wait = summaryCollection.waitSummaryAck(
+			containers[0].deltaManager.lastSequenceNumber,
+		);
+		const summaryResult = await summarizeNow(summarizer!);
+		assert(summaryResult.summaryVersion !== undefined, "summary result");
+		const ackedSummary = await wait;
+		assert(ackedSummary.summaryAck.contents.handle !== undefined, "summary acked");
+		return summaryResult.summaryVersion;
+	}
+
+	/**
+	 * Creates a pair of containers and initializes them with initial state
+	 * @returns collab space
+	 */
+	async function initialize() {
+		const { container, dataStores } = await createContainer();
+
+		// Create and setup a summary collection that will be used to track and wait for summaries.
+		summaryCollection = new SummaryCollection(container.deltaManager, createChildLogger());
+
+		const summarizerRes = await createSummarizerCore(container, loader!);
+		summarizer = summarizerRes.summarizer;
+		// containers.push(summarizerRes.container);
+
+		// Have a second container that follows passivley the first one
+		await addContainerInstance();
+
+		await provider.ensureSynchronized();
+
+		return dataStores;
+	}
+
+	it("Basic test", async () => {
```

Comment:
I would love to see if you could create another child `IDataStores` object. That would be really cool

Reply:
You mean under same parent? Or third level of nesting?

Reply:
The more the merrier, it's more of a curiosity thing to see what happens rather than a requirement for testing. So if you could do both, that would be awesome.